pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -297,6 +329,13 @@ func (r *ReconcilePravegaCluster) syncBookieSize(p *pravegav1alpha1.PravegaClust
 		if err != nil {
 			return fmt.Errorf("failed to sync pvcs of stateful-set (%s): %v", sts.Name, err)
 		}
+
+		p.Status.SetScalingCondition()
+		p.Status.ClearReadyCondition()
+		err = r.client.Update(context.TODO(), p)
```

Comment:
We are updating the CR object from multiple places just to update the status information. We should try to unify the CR status update in one place, e.g., in the new `reconcileClusterReadiness` method.

Reply:
You are right.